tests/query_test.go

```diff
@@ -40,3 +88,115 @@ func (s *QuerySuite) TestQuery() {
 		s.Equal("bar", store.MustFindOne(queryErr).Foo)
 	})
 }
+
+func (s *QuerySuite) TestFindById() {
+	store := NewQueryFixtureStore(s.db)
+
+	docName := "bar"
+	doc := NewQueryFixture(docName)
+	s.Nil(store.Insert(doc))
+
+	query := NewQueryFixtureQuery()
+	query.FindByID(doc.ID)
+	s.NotPanics(func() {
+		s.Equal(docName, store.MustFindOne(query).Foo)
+	})
+
+	queryManyId := NewQueryFixtureQuery()
+	queryManyId.FindByID(queryFixtures[1].ID, queryFixtures[2].ID)
+	count, err := store.Count(queryManyId)
+	s.Equal(2, int(count))
+	s.Nil(err)
+
+	notID := kallax.NewID()
+	queryErr := NewQueryFixtureQuery()
+	queryErr.FindByID(notID)
+	s.Panics(func() {
+		store.MustFindOne(queryErr)
+	})
+}
+
+func (s *QuerySuite) TestFindBy() {
+	store := NewQueryFixtureStore(s.db)
+	s.NotPanics(func() {
+		s.True(store.MustFindOne(NewQueryFixtureQuery().FindByStringProperty("StringProperty1")).Eq(queryFixtures[1]))
+	})
+	s.Panics(func() {
+		store.MustFindOne(NewQueryFixtureQuery().FindByStringProperty("NOT_FOUND"))
+	})
+
+	s.NotPanics(func() {
+		s.True(store.MustFindOne(NewQueryFixtureQuery().FindByBoolean(false)).Eq(queryFixtures[1]))
+	})
+	s.NotPanics(func() {
+		count, err := store.Count(NewQueryFixtureQuery().FindByBoolean(true))
+		s.Equal(int64(2), count)
+		s.Nil(err)
+	})
+
+	s.NotPanics(func() {
+		s.True(store.MustFindOne(NewQueryFixtureQuery().FindByInteger(kallax.Eq, 2)).Eq(queryFixtures[2]))
+	})
+	s.Panics(func() {
+		store.MustFindOne(NewQueryFixtureQuery().FindByInteger(kallax.Eq, 99))
+	})
+	s.NotPanics(func() {
+		count, err := store.Count(NewQueryFixtureQuery().FindByInteger(kallax.GtOrEq, 1))
+		s.Equal(int64(2), count)
+		s.Nil(err)
+	})
+	s.NotPanics(func() {
+		count, err := store.Count(NewQueryFixtureQuery().FindByInteger(kallax.Lt, 0))
+		s.Equal(int64(0), count)
+		s.Nil(err)
+	})
+}
+
+func (s *QuerySuite) TestGeneration() {
+	q := NewQueryFixtureQuery()
+	s.hasFindByMethod(q, "ID")
```

Comment:
how about a trable-driven tests?

```go
var cases = []struct{
    name string
    generated bool
}{
    {"ID", true},
    // ...
}

for _, c := range cases {
    if c.generated {
        s.hasFindByMethod(q, c.name)
    } else {
        s.hasNotFindByMethod(q, c.name)
    }
}
```

that way it's easier to add more cases and there is less repeated code